modules/xphoto/src/inpainting.cpp

```diff
@@ -66,6 +77,820 @@ namespace cv
 {
 namespace xphoto
 {
+
+    struct fsr_parameters
+    {
+        // default variables
+        int block_size = 16;
+        double conc_weighting = 0.5;
+        double rhos[4] = { 0.80, 0.70, 0.66, 0.64 };
+        double threshold_stddev_Y[3] = { 0.014, 0.030, 0.090 };
+        double threshold_stddev_Cx[3] = { 0.006, 0.010, 0.028 };
+        // quality profile dependent variables
+        int block_size_min, fft_size, max_iter, min_iter, iter_const;
+        double orthogonality_correction;
+        fsr_parameters(const int quality)
+        {
+            if (quality == xphoto::INPAINT_FSR_BEST)
+            {
+                block_size_min = 2;
+                fft_size = 64;
+                max_iter = 400;
+                min_iter = 50;
+                iter_const = 2000;
+                orthogonality_correction = 0.2;
+            }
+            else if (quality == xphoto::INPAINT_FSR_FAST)
+            {
+                block_size_min = 4;
+                fft_size = 32;
+                max_iter = 100;
+                min_iter = 20;
+                iter_const = 1000;
+                orthogonality_correction = 0.5;
+            }
+            else
+            {
+                CV_Error(cv::Error::StsBadArg, "Unknown quality level set, supported: FAST, COMPROMISE, BEST");
+
+            }
+        }
+    };
+
+
+    static void
+    icvBGR2YCbCr(const cv::Mat& bgr, cv::Mat& Y, cv::Mat& Cb, cv::Mat& Cr)
```

Comment:
Please use `cvtColor()` + `cv::split()` instead of manual function.

Reply:
Thanks a lot for your fast review!
I added a simple test with existing test data following your advice and tried to take all your comments into account.

I changed that and use ``cvtColor()`` now. 
The manual function was used to achieve the same value range for the YCbCr color space as in Matlab for a better comparison (the algorithm was ported from Matlab to C++ and the matlab color conversion covers a slighlty different value range as the result of ``cvtColor()``). 
So, the implementation might give slightly different results than those stated in the paper. I made a quick check for the Kodak image set, the resulting mean PSNR value differs by 0.04 dB.